### PR TITLE
ICU-22814 Run CIFuzz on PR

### PR DESCRIPTION
**Pull request by FrankYFTang**
CIFuzz ran on post merged since 7/20
https://github.com/unicode-org/icu/actions/workflows/cifuzz.yml

Turn to run on every PR

##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22814
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [X] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [X] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Comment by markusicu** [comment]
@roubert FYI

File: .github/workflows/cifuzz.yml [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
-  #    - '**.h'
-  # Only trigger by post merge for now.
+  pull_request:
+    branches: '**'
+    paths:
+      - '**.c'
+      - '**.cc'
+      - '**.cpp'
+      - '**.cxx'
+      - '**.h'
```
Comment (markusicu) [inline .github/workflows/cifuzz.yml:10]: Should we use the same paths regexes for pre-merge & post-merge triggering?

File: .github/workflows/cifuzz.yml [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
-  #    - '**.c'
-  #    - '**.cc'
-  #    - '**.cpp'
-  #    - '**.cxx'
-  #    - '**.h'
-  # Only trigger by post merge for now.
+  pull_request:
+    branches: '**'
+    paths:
+      - 'icu4c/**'
```
Comment (roubert) [inline .github/workflows/cifuzz.yml:6]: Is it really desireable to require this to be run also for changes that don't touch any source code?

**Review by roubert — COMMENTED** [review]
Wouldn't it be useful to also allow this to be triggered manually through `workflow_dispatch`?

**Comment by mihnita** [comment]
Would help if this would not fail.
It feels kind of wrong to approve the enablement of a test that fails :-)

**Comment by FrankYFTang** [comment]
> Would help if this would not fail. It feels kind of wrong to approve the enablement of a test that fails :-)

The failing is caused by the landing of https://github.com/unicode-org/icu/commit/6c9d39a08c469a755fcbec1719bcd0d3af217e7f and [roubert](https://github.com/roubert) is fixing that in https://github.com/unicode-org/icu/pull/3144 I need to rebase to pick up his fix.

**Review by mihnita — APPROVED** [review]

**Pull request merged**